cmd/pulumi-test-language/interface.go

```diff
@@ -0,0 +1,1025 @@
+		}
+		expectedDependencies := []plugin.DependencyInfo{}
+		if token.CoreVersion != "" {
+			expectedDependencies = append(expectedDependencies, plugin.DependencyInfo{
+				Name: "pulumi", Version: token.CoreVersion,
+			})
+		}
+		for _, provider := range test.providers {
+			pkg := string(provider.Pkg())
+			version, err := getProviderVersion(provider)
+			if err != nil {
+				return nil, err
+			}
+			expectedDependencies = append(expectedDependencies, plugin.DependencyInfo{
+				Name:    pkg,
+				Version: version.String(),
+			})
+		}
+		for _, expectedDependency := range expectedDependencies {
+			// We have to do some fuzzy matching by name here because the language plugin returns the name of the
+			// library, which is generally _not_ just the plugin name. e.g. "@pulumi/aws" for the nodejs aws library.
+
+			// When checking for version equality we _want_ to do a semver exact match but not all languages support
+			// semver, so we will allow a fuzzy match of the version as well.
+			versionsMatch := func(expected, actual string) bool {
+				if expected == actual {
+					return true
+				}
+				// Actual might be the empty string, some languages can't always return versions especially for local
+				// dependencies. In this case we treat it as matching as this is just supposed to be a best effort check.
+				if actual == "" {
+					return true
+				}
+
+				// Expected _will_ be a semver (because we got it from the provider version), but actual could be
+				// _anything_. We assume it will at least have the major.minor.patch part from the expected semver.
+				expectedSV := semver.MustParse(expected)
+				expectedSV.Pre = nil
+				expectedSV.Build = nil
+				expected = expectedSV.String()
+
+				return strings.Contains(actual, expected)
+			}
+
+			// found is the version we've found for this dependency, if any. We fuzzy match by name and then check version
+			// so this is just to give better error messages. For our main dependencies we should have a different version
+			// for every package, so the fuzzy check by name then exact check by version should be unique.
+			var found *string
+			for _, actual := range dependencies {
+				actual := actual
+
+				sanatize := func(s string) string {
+					return strings.ToLower(
+						strings.ReplaceAll(
+							strings.ReplaceAll(s, "_", ""),
+							"-", ""))
+				}
+
+				if strings.Contains(sanatize(actual.Name), sanatize(expectedDependency.Name)) {
+					found = &actual.Version
+					if versionsMatch(expectedDependency.Version, actual.Version) {
+						break
+					}
+				}
+			}
+
+			if found == nil {
+				return makeTestResponse("missing expected dependency " + expectedDependency.Name), nil
+			} else if !versionsMatch(expectedDependency.Version, *found) && expectedDependency.Name != "pulumi" {
+				return makeTestResponse(fmt.Sprintf("dependency %s has unexpected version %s, expected %s",
+					expectedDependency.Name, *found, expectedDependency.Version)), nil
+			}
```

Comment:
Here is where I made the change to exclude testing `pulumi` core sdk version, otherwise I didn't understand where the version was coming from since the nuget has v1.0.0